### PR TITLE
Section 2.8. Terminal

### PR DESCRIPTION
**Pull request by jurij-jukic**
1. I feel like the list of commands should be on a separate reference page. All docs in section 2 were conceptual, so it feel weird to have this here. Though, I'm not really sure where to put it.

2. 'For your scripts to be usable by the terminal, you must include a `pkg/scripts.json` file.'
- at which subfolder exactly? does this refer to when you have your own package? I found [this](https://github.com/kinode-dao/kinode/blob/main/kinode/packages/terminal/pkg/scripts.json) but thats a core package. it would be a good example, but it should somehow be made clear that that's not where it should be edited (?)

3. '`grantCapabilities` (Array): An array of `ProcessId`s as strings which represent which processes will be able to message this script back'
- is messaging the script back equivalent to having full capability? I assume in this context, yes, because it's just a script, but in general capabilities can be more elaborate (with an object that contains `issuer` and `params`)

### CODE REVIEW

**Comment by nick1udwig** [comment]
1. Good point. Please make an issue and we can address later
2. Best discussion of package directory format is [here](https://book.kinode.org/my_first_app/chapter_1.html#exploring-the-package) as far as I know. Yeah, I like the idea of linking terminal's scripts.json (with the notes/caveats you mention)
3. It is a `messaging` capability: if you don't have it and the process is not public you cannot message them. Feel free to make this sentence clearer

File: src/terminal.md [hunk 1]
```diff
@@ -83,16 +83,16 @@ For your scripts to be usable by the terminal, you must include a `pkg/scripts.j
 The JSON object in `scripts.json` describes the configuration for each script in your package.
 Each top-level key represents the path of the WASM module in your package, usually just `"myscript.wasm"`, `"echo.wasm"`, etc.
 
-The value for each module is an object that specifies the configuration for that particular module.
+Within this JSON object, for each key (WASM module) the value is an object that specifies the configuration for that particular module.
```
Comment (nick1udwig) [inline src/terminal.md:86]: WASM module -> process

File: src/terminal.md [hunk 2]
```diff
@@ -83,16 +83,16 @@ For your scripts to be usable by the terminal, you must include a `pkg/scripts.j
...
 - `public` (Boolean): Determines if the script is publicly accessible by other processes
 - `requestNetworking` (Boolean): Specifies whether the script will get networking capabilities
 - `requestCapabilities` (Array): An array that lists the capabilities requested by the script. Each element in the array can be either a string or an object. The string represents a `ProcessId` that this script will be able to message. When an object is used, it specifies a different kind of capability from `issuer` with `params` as an arbitrary json object.
-- `grantCapabilities` (Array): An array `ProcessId`s as strings which represent which processes will be able to message this script back
+- `grantCapabilities` (Array): An array of `ProcessId`s as strings which represent which processes will be able to message this script back
 
-Modules may not necessarily use all these fields. For instance, "m.wasm" only uses root, public, and requestNetworking, omitting requestCapabilities and grantCapabilities.
+WASM modules may not necessarily use all these fields. For instance, "m.wasm" only uses root, public, and requestNetworking, omitting requestCapabilities and grantCapabilities.
```
Comment (nick1udwig) [inline src/terminal.md:95]: 1. Split into one sentence per line
2. WASM modules -> Processes

**Review by nick1udwig — COMMENTED** [review]
Please check throughout for the following:
1. References to "modules" should be "processes"
2. change `camelCase` paramters to `snake_case` (we neglected to update this when code changes and so should do it now while we're here)

**Comment by jurij-jukic** [comment]
> 3. It is a `messaging` capability: if you don't have it and the process is not public you cannot message them. Feel free to make this sentence clearer
does this mean:
- if the process doesnt have a messaging capability granted by the script and that process is not public, it cannot message the script?
- also, why would a process need to be public to message the script?

**Comment by jurij-jukic** [comment]
> 2. change `camelCase` paramters to `snake_case` (we neglected to update this when code changes and so should do it now while we're here)
`PackageId` stays `camelCase`, correct?

**Comment by nick1udwig** [comment]
> `PackageId` stays `camelCase`, correct?

Yes -- rust types are like `FooBar`, but all our json fields should be like `hello_world`.

**Comment by nick1udwig** [comment]
> > 3. It is a `messaging` capability: if you don't have it and the process is not public you cannot message them. Feel free to make this sentence clearer
> >    does this mean:
> 
> * if the process doesnt have a messaging capability granted by the script and that process is not public, it cannot message the script?
> * also, why would a process need to be public to message the script?

Sorry, when I say a process, P, "has a cap" I mean "a process/module has granted P a cap. So what I meant here was: when you set `foo:bar:baz.os` in your scripts `grant_capabilities` field, you are giving `foo:bar:baz.os` the `messaging` capability for your script so that it can message your script back, e.g., to Respond with whatever result.

So in general, a process must either grant `messaging` for you to message them OR they can register as public in which case anyone can message them.

**Comment by jurij-jukic** [comment]
added a link to terminal syntax at the top of the page, please check

**Pull request merged**